document-portal/document-portal-fuse.c

```diff
@@ -455,6 +459,68 @@ xdp_physical_inode_ref (XdpPhysicalInode *inode)
   return inode;
 }
 
+static void
+xdp_physical_inode_use (XdpPhysicalInode *inode, int fd)
+{
+  gint old_use;
+
+retry_atomic_decrement1:
+  old_use = g_atomic_int_get (&inode->use_count);
+  if (old_use >= 1)
+    {
+      if (!g_atomic_int_compare_and_exchange ((int *) &inode->use_count, old_use, old_use + 1))
+        goto retry_atomic_decrement1;
+
+      close (fd);
+    }
+  else
+    {
+//      G_LOCK (physical_inodes);
+
+      if (!g_atomic_int_compare_and_exchange ((int *) &inode->use_count, old_use, old_use + 1))
+        {
+//          G_UNLOCK (physical_inodes);
+          goto retry_atomic_decrement1;
+        }
+      inode->fd = fd;
```

Comment:
Doesn't it leak ` inode->fd` here?

Reply:
we could add a check that it's -1 and if not close it. but this code path it is supposed to be closed.